src/index.js

```diff
@@ -686,17 +686,15 @@ export default function Vidy(opts) {
 			afterLogin = action;
 			return login.enter();
 		}
-		// current status
-		let cls = styles[action];
-		let value = has(actions, cls);
-		// apply reaction locally
+		// reset
 		actions.className = styles['vidy-btns'];
 		video._vlink._user_likes = video._vlink._user_angers = video._vlink._user_surprises = video._vlink._user_laughs = false;
-		toggle(actions, cls, (video._vlink[`_user_${action}s`] = !value));
-		setMetadata(video._vlink);
-		toggle(actions, styles['results'], !value); // results mode
-		// save to API
-		(value ? util.del : util.post)(`clips/${actions._clip}/reactions?type=${action}`);
+		// add reaction or toggle existing
+		let previous = !!actions.querySelector('.' + styles['active']) && actions.querySelector('.' + styles['active']).getAttribute('title').toLowerCase();
```

Comment:
Up to you if you wanna take this – I had to step thru it in a way that made sense to my peabrain 

```suggestion
let prev = actions.querySelector('.' + styles['active']);
prev = prev && prev.title.toLowerCase();
let isNew = prev && action !== prev;

// remove previous
if (isNew) video._vlink[`_user_${prev}s`] = false; 

// if no previous reaction or is new reaction ==> true
// otherwise, if is toggling same reaction ==> false 
video._vlink[`_user_${action}s`] = !prev || isNew; 

setMetadata(video._vlink); // update ui
(action === prev ? util.del : util.post)(`clips/${actions._clip}/reactions?type=${action}`);
```

(Don't think GH let's me down multi-line suggestion)